compiler/ml/typecore.ml

```diff
@@ -2826,13 +2826,25 @@ and type_expect_ ~context ?in_function ?(recarg = Rejected) env sexp ty_expected
         exp_env = env;
       }
   | Pexp_ifthenelse (scond, sifso, sifnot) -> (
+    (* TODO(attributes) Unify the attribute handling in the parser and rest of the compiler. *)
+    let is_ternary =
+      let rec has_ternary = function
+        | [] -> false
+        | ({Location.txt = "res.ternary"}, _) :: _ -> true
+        | _ :: rest -> has_ternary rest
+      in
+      has_ternary sexp.pexp_attributes
+    in
+    let return_context =
+      if is_ternary then Some TernaryReturn else Some IfReturn
+    in
```

Comment:
Putting in my TODO to unify all of these attribute names across the syntax and compiler. A lot of them should probably be proper AST nodes.